src/main/java/com/arpnetworking/clusteraggregator/models/CombinedMetricData.java

```diff
@@ -248,8 +247,8 @@ private static Quantity getQuantity(final Messages.StatisticRecord statisticReco
 
         @SuppressWarnings("unchecked")
         private static <T> T deserialzeSupportingData(final Messages.StatisticRecord record) {
-            if (!record.hasSupportingData()) {
-                throw Throwables.propagate(new IllegalArgumentException("no supporting data found"));
+            if (record.getSupportingData() == null) {
+                throw new RuntimeException(new IllegalArgumentException("no supporting data found"));
```

Comment:
Yes, but I didn't want to change the level of wrapping in this PR.  It's something we can look into later.